### PR TITLE
Use tokio to replace async_std for plugins and backends.

### PR DESCRIPTION
**Pull request by evshary**
Related to https://github.com/eclipse-zenoh/zenoh/issues/1257

### CODE REVIEW

**Comment by Mallets** [comment]
@wyfo do you mind reviewing this PR?

**Comment by YuanYuYuan** [comment]
BTW, if we decide to use another static Tokio runtime in the plugin and consequently remove the async-std dependency in zenoh-util, we can eliminate the outdated TimedEvent since it was used before Tokio-utils was available.

**Comment by Mallets** [comment]
> BTW, if we decide to use another static Tokio runtime in the plugin and consequently remove the async-std dependency in zenoh-util, we can eliminate the outdated TimedEvent since it was used before Tokio-utils was available.

We need to double check first wether `TimedEvent` is used in any plugin.

**Comment by YuanYuYuan** [comment]
A quick search over our eclipse project shows we use `TimedEvent` in the zenoh-plugin-dds, zenoh-plugin-ros2dds, and zenoh-backend-influxdb. And I agree that we should open another PR dedicated to cleaning this last async-std dependence in zenoh.

File: plugins/zenoh-plugin-storage-manager/tests/wildcard.rs [hunk 1]
```diff
@@ -195,6 +193,7 @@ async fn test_wild_card_in_order() {
 
 #[test]
 fn wildcard_test() {
-    task::block_on(async { test_wild_card_in_order().await });
-    // task::block_on(async { test_wild_card_out_of_order() });
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async { test_wild_card_in_order().await });
+    // rt.block_on(async { test_wild_card_out_of_order() });
```
Comment (Mallets) [inline plugins/zenoh-plugin-storage-manager/tests/wildcard.rs:198]: Dead code.
```suggestion
```

File: commons/zenoh-core/src/macros.rs [hunk 2]
```diff
@@ -157,7 +157,7 @@ macro_rules! to_u64 {
 }
 
 // This macro allows to spawn the right amount of threads in the
-// async_std executor
+// tokio executor
```
Comment (Mallets) [inline commons/zenoh-core/src/macros.rs:160]: The `zasync_executor_init` is specific to `async-std` and it is needed for tests to run on the CI. If we are moving to *tokio* then `zasync_executor_init` shouldn't be needed anymore. This change should be reverted and any call to `zasync_executor_init` should be removed from all the tests ported to *tokio*.

**Comment by Mallets** [comment]
A broader discussion should happen on wether every plugin has its own executor or a common executor can be used when statically linking the plugins.

**Comment by Mallets** [comment]
Discussion happened on Discord and there is a common agreement to go for a single runtime in case of static linking. 

To deal with *static* vs *dynamic* linking and symbols re-export `S3` backend uses the [dynamic_plugin](https://github.com/eclipse-zenoh/zenoh-backend-s3/blob/0973237a4348528b2420d8729894618ed5807273/Cargo.toml#L37) feature to selectively activate ad-hoc support for each target. We should explore that approach and to eventually make it common to all plugins.

File: plugins/zenoh-plugin-rest/src/lib.rs [hunk 3]
```diff
@@ -245,8 +257,9 @@ impl Plugin for RestPlugin {
 
         let conf: Config = serde_json::from_value(plugin_conf.clone())
             .map_err(|e| zerror!("Plugin `{}` configuration error: {}", name, e))?;
-        let task = async_std::task::spawn(run(runtime.clone(), conf.clone()));
-        let task = async_std::task::block_on(task.timeout(std::time::Duration::from_millis(1)));
+        let task = TOKIO_RUNTIME.spawn(run(runtime.clone(), conf.clone()));
```
Comment (Mallets) [inline plugins/zenoh-plugin-rest/src/lib.rs:260]: Here we should try to get the current runtime.

File: plugins/zenoh-plugin-rest/src/lib.rs [hunk 4]
```diff
@@ -245,8 +257,9 @@ impl Plugin for RestPlugin {
...
+        let task = TOKIO_RUNTIME
```
Comment (Mallets) [inline plugins/zenoh-plugin-rest/src/lib.rs:261]: Here we should try to get the current runtime.

**Review by Mallets — APPROVED** [review]

**Pull request merged**